:construction: create insert-enpoint

Evandro-Ladislau opened the pull request:
![Captura de tela de 2025-01-22 20-59-31](https://github.com/user-attachments/assets/3b2d5057-82ee-4d99-b2a6-d3a699aa26f9)

Evandro-Ladislau committed:
:construction: create insert-enpoint

AurelioNM commented on src/main/java/br/plataforma/eb/Models/CustomerModel.java:
```diff
@@ -24,6 +24,10 @@ public class CustomerModel {
     @Getter
     private Boolean isActive;
 
+    public CustomerModel(){
```
Esse construtor vazio aqui seria pra que?

Evandro-Ladislau replied:
O Jackson não estava conseguindo  criar uma instância da model com o body da requisição, dai explicitei o construtor padrão para isso. 
Teria uma forma melhor para fazer isso?

AurelioNM commented on src/main/java/br/plataforma/eb/Routes/Api.java:
```diff
@@ -1,5 +1,48 @@
+        app.post("/insert", ctx -> {
+            try {
+                var customer = ctx.bodyAsClass(CustomerModel.class);
+                CustomerModel customerModel = new CustomerModel(customer.getName(), customer.getSurname(), customer.getEmail(), customer.getBirthDate());
```
Essa linha 34 ta me parecendo redundante pq na linha de cima vc meio q ja ta tentando converter o body como a model ne?
Eu imagino que vc tenha feito isso pro Jackson fazer o parsing do json dentro da model
Mas acho que colocando os decorators do jackson nas propriedades da classe, resolve o problema

Aqui no trampo é comum agnt colocar esse decorator na class
`@JsonNaming(PropertyNamingStrategies.LowerCamelCaseStrategy::class)`
é coisa do `fasterxml.jackson.databind.annotation.JsonNaming
`

Evandro-Ladislau replied:
aaah, foi exatamento por conta disso que eu fiz.
então com os decorators vai resolver. vou melhorar isso.

AurelioNM commented on src/main/java/br/plataforma/eb/Services/CustomerService.java:
```diff
@@ -16,6 +16,7 @@ public CustomerModel insert(CustomerModel customer) throws SQLException {
 
         try {
             logger.info("Calling the repository layer");
+            System.out.println("Model: " + customer.getId() + " " + customer.getName() + " " + customer.getBirthDate() + " " + customer.getEmail() + " " + customer.getCreatedAt());
```
Precisa colocar campo a campo mesmo? 
Acho q vc pode criar uma funcao toString dentro da Model, é comum termos ela pra por exemplo colocar so customer dentro do log e por debaixo dos panos o .toString() é chamado, retornando uma string q represente a classe

E esse log deveria ta junto do primeiro  `logger.info`, não num println separado

Evandro-Ladislau replied:
Realmente, inclusive se não me engano lombok cria esse método.
Isso ai eu tinha colocado só para testar e esqueci de tirar, mas como é interessante vou mantar no logger.info com o toString.

AurelioNM commented on src/main/java/br/plataforma/eb/Routes/Api.java:
```diff
@@ -1,5 +1,48 @@
+        logger.info("Calling endpoint insert");
+
+        app.post("/insert", ctx -> {
+            try {
```
Pode colocar o log `logger.info("Calling endpoint insert");` dentro do try

Evandro-Ladislau replied:
certo!

Evandro-Ladislau committed:
:construction: create insert-enpoint: - Refatorações baseadas no feedback do PR.
- Adicionada classe JacksonConfig para desserialização de datas.
- Implementado toString em CustomerModel com Lombok.

Evandro-Ladislau committed:
:wastebasket: create-insert-endpoint - remove class JacksonConfig

AurelioNM commented on src/main/java/br/plataforma/eb/Models/CustomerModel.java:
```diff
@@ -16,15 +22,19 @@ public class CustomerModel {
 
-    public CustomerModel(String name, String surname, String email, LocalDate birthDate){
+    @JsonCreator
+    public CustomerModel(@JsonProperty("name") String name,  @JsonProperty("surname") String surname, @JsonProperty("email") String email,  @JsonProperty("birthDate")  LocalDate birthDate){
```
Isso ae, esse decorator ae msm
Mas a ideia seria usar ele direto no atributo, ao inves do construtor
Por exemplo, chegar ali na linha 23 e deixar assim
@Getter
@JsonProperty("name")
private String name

AurelioNM commented:
Boa, ta ótimo!

Evandro-Ladislau committed:
:construction: create insert-enpoint: - Refatorações baseadas no feedback do PR.
- Removi método generateULID.
